### PR TITLE
user creation

### PR DESCRIPTION
**Pull request by ps-eqsdigital**
user creation
return code 400 if user lacks data or email is invalid
sends email with generated password (random)

### CODE REVIEW

File: api/Business/BusinessModel.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Business
+{
+    internal class BusinessModel
```
Comment (tentone) [inline api/Business/BusinessModel.cs]: Empty class, any reason?

File: api/.vs/api/v17/.suo
Comment (tentone) [inline api/.vs/api/v17/.suo:1]: Add .vs to gitignore and remove from repository

File: api/Business/BusinessObjects/UserBusinessObject.cs [hunk 2]
```diff
@@ -40,5 +45,68 @@ public async Task<OperationResult> Update(Guid uuid, User record)
...
+            mail.Subject = "User password";
+            mail.Body = "Your password is " + password;
+
+           
+            // Create a SMTP client and send the email
+            SmtpClient smtpClient = new SmtpClient("smtp.sapo.pt");
+            smtpClient.Port = 587;
+            smtpClient.Credentials = new NetworkCredential("unoonboarding@sapo.pt", "Eqs_2023");
+            smtpClient.EnableSsl = true;
+
```
Comment (tentone) [inline api/Business/BusinessObjects/UserBusinessObject.cs]: Store SMTP configuration in config file

File: api/Business/bin/Debug/net6.0/DataAccess.dll
Comment (tentone) [inline api/Business/bin/Debug/net6.0/DataAccess.dll:1]: Remove bin

File: api/Data/obj/Debug/net6.0/Data.GeneratedMSBuildEditorConfig.editorconfig
Comment (tentone) [inline api/Data/obj/Debug/net6.0/Data.GeneratedMSBuildEditorConfig.editorconfig:1]: remove all Debug build code

**Review by tentone — CHANGES_REQUESTED** [review]

File: api/Business/BusinessModel.cs (hunk 1 above)
Comment (ps-eqsdigital) [reply api/Business/BusinessModel.cs]: created by mistake

File: api/Business/BusinessObjects/UserBusinessObject.cs (hunk 2 above)
Comment (ps-eqsdigital) [reply api/Business/BusinessObjects/UserBusinessObject.cs]: added to appsettings.json

File: api/Business/BusinessObjects/UserBusinessObject.cs [hunk 3]
```diff
@@ -40,5 +47,86 @@ public async Task<OperationResult> Update(Guid uuid, User record)
                 await _genericDataAccessObject.UpdateAsync<User>(user);
             });
         }
+
+        public async Task<OperationResult<CreateUserBusinessModel>> Insert(User record)
+        {
+            return await ExecuteOperation(async () =>
+            {
+                if (!IsValidEmail(record.Email!) | record.Name.IsNullOrEmpty() | record.Picture.IsNullOrEmpty() | record.Phone.IsNullOrEmpty())
```
Comment (js-eqsdigital) [inline api/Business/BusinessObjects/UserBusinessObject.cs]: should be || when using or

File: api/Business/BusinessObjects/UserBusinessObject.cs [hunk 4]
```diff
@@ -40,5 +47,86 @@ public async Task<OperationResult> Update(Guid uuid, User record)
...
+                {
+                    throw new Exception();
+                }
+                string password = GenerateRandomPassword();
+                record.Password = password;
+                Console.WriteLine("here");
```
Comment (js-eqsdigital) [inline api/Business/BusinessObjects/UserBusinessObject.cs]: ?

**Review by js-eqsdigital — CHANGES_REQUESTED** [review]

File: api/Business/BusinessObjects/UserBusinessObject.cs (hunk 4 above)
Comment (ps-eqsdigital) [reply api/Business/BusinessObjects/UserBusinessObject.cs]: debug stuff, forgot to remove

File: api/Business/BusinessObjects/UserBusinessObject.cs (hunk 3 above)
Comment (ps-eqsdigital) [reply api/Business/BusinessObjects/UserBusinessObject.cs]: changed

**Pull request merged**